mungegithub/mungers/issue-creator.go

```diff
@@ -0,0 +1,396 @@
+func (creator *IssueCreator) AddFlags(cmd *cobra.Command, config *github.Config) {
+	cmd.Flags().StringVar(&creator.ownerPath, "test-owners-csv", "", "file containing a CSV-exported test-owners spreadsheet")
+	cmd.Flags().IntVar(&creator.maxSIGCount, "maxSIGs", 3, "The maximum number of SIG labels to attach to an issue.")
+	cmd.Flags().IntVar(&creator.maxAssignees, "maxAssignees", 3, "The maximum number of users to assign to an issue.")
+}
+
+//Munge updates the IssueCreator's cache of issues if the munge object provided is an issue authored by the currently authenticated user.
+func (creator *IssueCreator) Munge(obj *github.MungeObject) {
+	//ignore pull requests
+	if obj.IsPR() {
+		return
+	}
+	//ignore issues that were created by other users
+	if *obj.Issue.User.Login != creator.authorName {
+		return
+	}
+	creator.allIssues[*obj.Issue.Number] = obj //adds or updates
+}
+
+//filterValidLabels extracts the labels that are valid for the current repo from a set of labels.
+//If a candidate label is not valid, an error is logged, but execution continues and the invalid
+//label is ignored.
+func (creator *IssueCreator) filterValidLabels(candidates []string) []string {
+	if creator.validLabels == nil {
+		//failed to fetch list of valid labels for this repo so trust that labels are valid
+		return candidates
+	}
+	var validCandidates []string
+	for _, l := range candidates {
+		found := false
+		for _, validLabel := range creator.validLabels {
+			if l == *validLabel.Name {
+				found = true
+				break
+			}
+		}
+		if !found {
+			glog.Errorf("The label '%s' is invalid for the repo '%s/%s'. Dropping label.", l, creator.config.getOrg(), creator.config.getProject())
+		} else {
+			validCandidates = append(validCandidates, l)
+		}
+	}
+	return validCandidates
+}
+
+//Sync checks to see if an issue is already on github and tries to create a new issue for it if it is not.
+func (creator *IssueCreator) Sync(issue Issue) {
+	//first look for existing issues with this ID
+	id := issue.ID()
+	var closedIssues []*github.MungeObject
+	for _, obj := range creator.allIssues {
+		if strings.Contains(*obj.Issue.Body, id) {
+			switch *obj.Issue.State {
+			case "open":
+				//if an open issue is found with the ID then the issue is already synced
+				return
+			case "closed":
+				closedIssues = append(closedIssues, obj)
+			default:
+				glog.Errorf("Unrecognized issue state for issue #%d. Ignoring this issue.\n", *obj.Issue.Number)
+			}
+		}
+	}
+	//if we get here then no open issues exist for the ID
+	body := issue.Body(closedIssues)
+	if body == "" {
+		//Issue indicated that it should not be synced
+		return
+	}
+	if !strings.Contains(body, id) {
+		glog.Fatalf("Programmer error: The following body text does not contain id '%s'.\n%s\n", id, body)
+	}
+
+	title := issue.Title()
+	owners := issue.Owners()
+	labels := issue.Labels()
+	if prio, ok := issue.Priority(); ok {
+		labels = append(labels, "priority/P"+strconv.Itoa(prio))
```

Comment:
Done. Priorities are not currently used but are now of type string instead of an int.